pkg/apis/core/validation/controllerregistration.go

```diff
@@ -84,5 +118,15 @@ func ValidateControllerRegistrationSpecUpdate(new, old *core.ControllerRegistrat
 		return allErrs
 	}
 
+	kindTypeToPrimary := make(map[string]*bool, len(old.Resources))
+	for _, resource := range old.Resources {
+		kindTypeToPrimary[resource.Kind+resource.Type] = resource.Primary
+	}
+	for i, resource := range new.Resources {
+		if primary, ok := kindTypeToPrimary[resource.Kind+resource.Type]; ok {
+			allErrs = append(allErrs, apivalidation.ValidateImmutableField(resource.Primary, primary, fldPath.Child("resources").Index(i).Child("primary"))...)
```

Comment:
I'm not sure if this validation makes sense because an operator can easily circumvent it by removing the resource from the list and adding it again afterwards. WDYT?

Reply:
IMO it makes sense as it prevents obvious misconfiguration. If somebody uses the workaround you mentioned then it's probably/hopefully a conscious decision and the person is aware of the consequences.

Reply:
Okay, fair enough